### PR TITLE
Feature/dyslexia swap feature

### PR DESCRIPTION
**Pull request by ArkajyotiChakraborty**
For issue [344](https://github.com/JohnSnowLabs/nlptest/issues/344)

-I've added Google style docstrings to my code. [here](https://github.com/ArkajyotiChakraborty/nlptest/blob/29b9871a269b61628e2d4e38abd46d7cd1888fe7/nlptest/transform/robustness.py#L700)
-I have added tests to cover my changes. [here](https://github.com/ArkajyotiChakraborty/nlptest/blob/29b9871a269b61628e2d4e38abd46d7cd1888fe7/tests/test_robustness.py#L139)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+pydantic==1.10.6
+spacy
+johnsnowlabs==4.3.5
+plotly
+asyncio
+nest-asyncio
+jsonlines
+openai
+langchain
+spacy
```
Comment (alytarik) [inline requirements.txt:16]: we dont need spacy and en_core_web_sm here, they are optional to install

**Review by alytarik — CHANGES_REQUESTED** [review]

File: nlptest/transform/robustness.py [hunk 2]
```diff
@@ -692,3 +693,52 @@ def search_contraction(text):
                     sample.transformations = transformations
             sample.category = "robustness"
         return sample_list
+
+class Dyslexia_Word_Swap(BaseRobustness):
```
Comment (alytarik) [inline nlptest/transform/robustness.py:697]: Please use TitleCase for class name.

File: nlptest/transform/robustness.py [hunk 3]
```diff
@@ -692,3 +693,52 @@ def search_contraction(text):
...
+            return new_sentence
+        
+        for sample in sample_list:
+            if "task" in sample.__annotations__:
+                sample.perturbed_question = generate(sample.original_question)
+                if "perturbed_context" in sample.__annotations__:
+                    sample.perturbed_context = generate(sample.original_context)
+            else:
+                sample.test_case = generate(sample.original)
+                if sample.test_case==None: print(sample)
```
Comment (alytarik) [inline nlptest/transform/robustness.py:742]: Was this for debugging? I think you can also remove this

**Review by alytarik — CHANGES_REQUESTED** [review]
please check the 3 comments

File: requirements.txt (hunk 1 above)
Comment (ArkajyotiChakraborty) [reply requirements.txt:16]: I have made the other two changes as requested and updated the Class name in the robustness.py file and in test_robustness.py where the class name is being called. The original code with which the issue was associated used spacy and en_core_web_sm as it requires in the further loop so what should I do about that.

**Review by alytarik — APPROVED** [review]

**Comment by alytarik** [comment]
thanks for your work @ArkajyotiChakraborty, we will adapt your code into the library

**Pull request merged**